File: AndroidClient/app/src/main/java/androidclient/automacaoz/raspberry/bruno/azandroidclient/AppClasses/SensorClass.java
package androidclient.automacaoz.raspberry.bruno.azandroidclient.AppClasses;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Bruno on 24/06/2017.
 */

public class SensorClass {
    private final String TAG = "SENSOR_CLASS";
    String sensorName;
    String sensorAction;

    public SensorClass(String sensorName, String sensorAction) {
        this.sensorName = sensorName;
        this.sensorAction = sensorAction;
    }

    public SensorClass(String jsonObj){
        JSONObject json;
        try {
            json = new JSONObject(jsonObj);
        } catch (Exception e){
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return;
        }


        try {
            this.sensorName = json.getString("sensorName");
            this.sensorAction = json.getString("sensorAction");
        } catch (JSONException e) {
            e.printStackTrace();
        }

    }

    public String parseToJson(){
        JSONObject json = new JSONObject();
        try {
            json.put("sensorName", sensorName);
            json.put("sensorAction", sensorAction);
        } catch (JSONException e) {
            e.printStackTrace();
            Log.i(TAG, "Erro ao criar o JSON");
            return null;
        }

        return json.toString();
    }

    public String getSensorName() {return sensorName;}
    public void setSensorName(String sensorName) {this.sensorName = sensorName;}
    public String getSensorAction() {return sensorAction;}
    public void setSensorAction(String sensorAction) {this.sensorAction = sensorAction;}
}

File: RaspberryServer2/src/AppClasses/SensorClass.cpp
#include "SensorClass.h"

SensorClass::SensorClass(string _sensorName, string _sensorAction){
    this->sensorName = _sensorName;
    this->sensorAction = _sensorAction;
}

SensorClass::SensorClass(){}

SensorClass::~SensorClass(){}


string SensorClass::parseToJson(){
    DynamicJsonBuffer jsonBuffer;

    JsonObject& root = jsonBuffer.createObject();
    root["sensorName"] = this->sensorName;
    root["sensorAction"] = this->sensorAction;

    string json;
    root.printTo(json);
    return json;
}

bool SensorClass::createFromJson(string json){
    DynamicJsonBuffer jsonBuffer;
    JsonObject& root = jsonBuffer.parseObject(json);

    //TODO: tirar essa gambs de primeiro usar um char*
    const char* tmp = root["sensorName"];
    const char* tmp2 = root["sensorAction"];

    if ((tmp == NULL) || (tmp2 == NULL)) {
        return false;
    }
    this->sensorName = string(tmp);
    this->sensorAction = string(tmp2);

    return true;
}




void SensorClass::setSensorName(string _sensorName){
    this->sensorName = _sensorName;
}
void SensorClass::setSensorAction(string _sensorAction){
    this->sensorAction = _sensorAction;
}
string SensorClass::getSensorName(){
    return this->sensorName;
}
string SensorClass::getSensorAction(){
    return this->sensorAction;
}



